### PR TITLE
cephfs: add SelectFilesystem, a 2nd fs to micro-osd.sh etc, and tests

### PR DESCRIPTION
**Pull request by phlogistonjohn**
This PR updates the micro-osd.sh script to host an alternate cephfs filesystem, on pacific and later. 
The entrypoint.sh and makefile are updated to plumb information about the alternate fs to the test suite.
The SelecFilesystem function is added implementing ceph_select_filesystem.
A group of tests, gated by the name of the alternate filesystem being provided, is added, along with API status updates.

Fixes: https://github.com/ceph/go-ceph/issues/262
Fixes: https://github.com/ceph/go-ceph/issues/818
Original-Version-By: "Arvid E. Picciani" <arvid@kraud.cloud>
Original-PR-URL: https://github.com/ceph/go-ceph/pull/819

## Checklist
- [x] Added tests for features and functional changes
- [x] Public functions and types are documented
- [x] Standard formatting is applied to Go code
- [x] Is this a new API? Added a new file that begins with `//go:build ceph_preview`
- [x] Ran `make api-update` to record new APIs

New or infrequent contributors may want to review the go-ceph [Developer'sGuide](https://github.com/ceph/go-ceph/blob/master/docs/development.md) including the section on how we track [API Status](https://github.com/ceph/go-ceph/blob/master/docs/development.md#api-status) and the [API Stability Plan](https://github.com/ceph/go-ceph/blob/master/docs/api-stability.md).

The go-ceph project uses mergify. View the [mergify command guide](https://docs.mergify.com/commands/#commands) for information on how to interact with mergify. Add a comment with `@Mergifyio` `rebase` to rebase your PR when github indicates that the PR is out of date with the base branch.

### CODE REVIEW

**Comment by phlogistonjohn** [comment]
Keeping it in draft to run the CI on the various branches which I haven't done locally.

**Comment by phlogistonjohn** [comment]
Mystery failures on nautilus and octopus. Failures in cephfs admin tests that assume there's exactly one fs. The latter should be easy to fix. I'll investigate the former soon. Watch this space!  ;-)

File: cephfs/admin/volume_test.go
Comment (ansiwen) [inline cephfs/admin/volume_test.go:16]: Just a nit: I bet `Contains` implies a `len()` check. Before it was necessary, because the `Equal()` directly dereferenced `vl[0]`.

File: cephfs/admin/volume_test.go
Comment (ansiwen) [inline cephfs/admin/volume_test.go:402]: another nit (and the code was like that before): wouldn't that be a case for `require.NotEqual()`? I never noticed, that `assert` returns a bool, and I wonder why.

File: cephfs/select_fs.go
Comment (ansiwen) [inline cephfs/select_fs.go:8]: Where does that come from? Is that documented in the libcephfs docs?

**Review by ansiwen — APPROVED** [review]
LGTM, set DNM label to give you a change to read my nit comments.

File: cephfs/select_fs.go
Comment (phlogistonjohn) [reply cephfs/select_fs.go:8]: I think it was there before we started maintaining the project. I copy and pasted this part from the other cephfs go file(s). We can research if it's strictly needed but I'd keep it for now for consistencies sake.

File: cephfs/admin/volume_test.go
Comment (phlogistonjohn) [reply cephfs/admin/volume_test.go:402]: You could use require for this but I didn't feel like mixing both assert and require in this case.
I'm pretty sure using the assert calls this way is documented somewhere at testify.

File: cephfs/admin/volume_test.go
Comment (phlogistonjohn) [reply cephfs/admin/volume_test.go:16]: Reasonable. But I think it does no harm. I'll change it only if I need to touch the file for another reason.

File: docs/api-status.json
Comment (anoopcs9) [inline docs/api-status.json:319]: Aren't we missing spaces in the first sentence of comment?

```
"comment": "Select file system to be mounted.
```

**Review by anoopcs9 — COMMENTED** [review]
All good, just a minor comment(see below).

File: docs/api-status.json
Comment (phlogistonjohn) [reply docs/api-status.json:319]: Go documentation style requires the first word of a "doc comment" to be the name of the function or type you are documenting. Therefore "SelectFilesystem" since that's the name of the function. It also encourages you to make clear english sentences. THere are a bunch of examples in the go stdlib. See also https://go.dev/doc/comment#func
I could change it to "SelectFilesystem selects the file system to be mounted. ..." I just figured most people would mentally insert the space.  Let me know if you feel this still needs to be changed.

Comment (anoopcs9) [reply docs/api-status.json:319]: Oh..all this while I failed to notice such a requirement. But I would still hang on to the point that encourages to make clear(and complete) English sentences(if you could).

Comment (phlogistonjohn) [reply docs/api-status.json]: OK, I wasn't so wedded to the previous wording so I updated it.

**Comment by phlogistonjohn** [comment]
I updated the doc comment. To save a little time I already set the version numbers in the API status by running `make api-fix-versions` myself. *IF* we don't merge before the release I'll manually adjust the numbers myself. I just hope this isn't needed (nudge, nudge :-))


This has dismissed the previous reviews, so @ansiwen @anoopcs9 please take another look. Thanks.

**Review by anoopcs9 — COMMENTED** [review]
lgtm, thanks.

File: docs/api-status.json
Comment (anoopcs9) [reply docs/api-status.json]: Thanks.

**Review by anoopcs9 — APPROVED** [review]

**Review by ansiwen — APPROVED** [review]

**Comment by ansiwen** [comment]
@mergifyio rebase

**Comment by ansiwen** [comment]
@Mergifyio refresh

**Comment by ansiwen** [comment]
Mergify is on vacation?

**Comment by phlogistonjohn** [comment]
Looks like we need to make a change to mergify config. It shows the following when I look at the status:
```


    The configuration uses the deprecated rebase_fallback mode of the queue and/or merge action.
    A brownout is planned on February 13th, 2023.
    This option will be removed on March 13th, 2023.
    For more information: https://docs.mergify.com/actions/queue/ or https://docs.mergify.com/actions/merge/

```
So we're hitting this brownout I guess.  As I would like to expidite this PR for the review. I am going to hit the merge button as I have two approvals for this PR. I will follow up with a fix for mergify config later today, but that won't be desired for the release this week.

**Pull request merged**
